xline/src/server/xline_server.rs

```diff
@@ -303,125 +248,103 @@ where
+        let ce = CommandExecutor::new(
+            Arc::clone(&kv_storage),
+            Arc::clone(&auth_storage),
+            Arc::clone(&lease_storage),
+            Arc::clone(&persistent),
+            Arc::clone(&index_barrier),
+            Arc::clone(&id_barrier),
+            header_gen.revision_arc(),
+            Arc::clone(&auth_revision_gen),
+        );
+        let snapshot_allocator: Box<dyn SnapshotAllocator> = match self.storage_cfg {
+            StorageConfig::Memory => Box::new(MemorySnapshotAllocator),
+            StorageConfig::RocksDB(_) => Box::new(RocksSnapshotAllocator),
             #[allow(clippy::unimplemented)]
             _ => unimplemented!(),
         };
-        (
+
+        Ok((
             KvServer::new(
-                Arc::clone(&self.kv_storage),
-                Arc::clone(&self.auth_storage),
-                Arc::clone(&self.index_barrier),
-                Arc::clone(&self.id_barrier),
-                self.range_retry_timeout,
-                Arc::clone(&self.client),
-                self.id.clone(),
+                kv_storage,
+                Arc::clone(&auth_storage),
+                Arc::clone(&index_barrier),
+                Arc::clone(&id_barrier),
+                *self.server_timeout.range_retry_timeout(),
+                Arc::clone(&client),
+                self.cluster_info.get_self_id().clone(),
             ),
             LockServer::new(
-                Arc::clone(&self.client),
-                Arc::clone(&self.id_gen),
-                self.id.clone(),
-                address,
+                Arc::clone(&client),
+                Arc::clone(&id_gen),
+                self.cluster_info.get_self_id().clone(),
+                self.cluster_info.self_address().to_owned(),
             ),
             LeaseServer::new(
-                Arc::clone(&self.lease_storage),
-                Arc::clone(&self.auth_storage),
-                Arc::clone(&self.client),
-                self.id.clone(),
-                Arc::clone(&self.id_gen),
-                self.all_members.clone(),
+                lease_storage,
+                Arc::clone(&auth_storage),
+                Arc::clone(&client),
+                id_gen,
+                Arc::clone(&self.cluster_info),
             ),
             AuthServer::new(
-                Arc::clone(&self.auth_storage),
-                Arc::clone(&self.client),
-                self.id.clone(),
+                auth_storage,
+                client,
+                self.cluster_info.get_self_id().clone(),
             ),
-            WatchServer::new(Arc::clone(&self.watcher), Arc::clone(&self.header_gen)),
-            MaintenanceServer::new(Arc::clone(&self.persistent), Arc::clone(&self.header_gen)),
-            curp_server,
-        )
+            WatchServer::new(watcher, Arc::clone(&header_gen)),
+            MaintenanceServer::new(persistent, header_gen),
+            CurpServer::new(
+                Arc::clone(&self.cluster_info),
```

Comment:
Same above

Reply:
IMO, we should clone here otherwise the `cluster_info` will be moved and corrupt the `XlineServer`

Reply:
Sry, I ignored the `self`